### DISCUSSION TITLE
What were/are the reasons for putting Space and Return on the left thumbs?

### DISCUSSION THREAD

**Discussion by jeffsf** [urob/zmk-config#75] [Q&A] [answered]
Trying out the Space and Return on the left thumbs is a struggle for me to retrain my right-thumb habit. Rather than simply give up and swap to the right, I'm wondering if there is a good enough reason to keep them on the left.

"Only because I was a left-thumb spacer out of habit" is a completely acceptable answer if that was the case.

**Comment by urob** [comment, marked as answer]
Great question! In parts, it's what I'm used to. But while switching up keyboard habits makes for a few really awkward days, it's also pretty quick in my experience to get used to (I not so long ago [changed](https://github.com/urob/zmk-config/commit/473be6331c6b29687a70f37957e8760f72b8f8f6) from `GACS` to `GASC` home-row mods, and couldn't image otherwise now).

The deeper reason is that it's the only solution I could come up with to the following constrained optimization program:

1. I only want 2 keys for each thumb to minimize lateral motion
2. `smart-shift` should be the tap & hold-action on one of the two home thumbs
3. `space` should be the tap-action on the other home thumb
4. `enter` should be the tap-action on the secondary thumb adjacent to `shift`
5. `smart-num` should be the tap & hold-action on a thumb
6. The `nav` layer should be the hold-action of one of my home thumbs (it's my most frequently used layer)
7. Layer activators should be opposite of the primary keys of the activated layer (Miryoku-style)
8. The navigation keys should be on the right side (works great in Vim)
9. The numbers should be on the left side (so I can use them for Excel or Sudoku while navigating a mouse with the right hand)

Having `space` on the left home thumb and `shift` on the right home thumb is the unique solution to `1-9`. In fact, the system is over-identified, so it's somewhat fortunate that there is a solution at all.

## Proof 1

10. `1-5` imply that `smart-num` has to be the secondary thumb adjacent to `shift`
11.  `7, 9 & 10` imply that `smart-num` has to be the secondary thumb on the right side
12. `10 & 11` imply that `shift` has to be the home thumb on the right side
13. `1, 3 & 12` imply that `space` has to be the home thumb on the left side
14. `4 & 13` imply `enter` should be the secondary thumb on the left side

## Proof 2

16. `6 & 8` imply that `nav` has to be the hold-action of my left home thumb
17. `2 & 14` imply that `smart-shift` has to be the tap-action on my right home thumb
18. `1, 3 & 16` imply that `space` has to be the tap-action on my left home thumb
19. `4 & 18` imply `enter` should be the secondary thumb on the left side

(sorry for going full nerd here, I show myself out 🤪)

**Reply by urob** [reply]
As you can see, `8 & 9` are the import inputs that pin down which keys are assigned left & which are assigned right. If they don't apply to you, you could try mirroring the entire layout. But if some of the other inputs apply to you, this may require swapping lots of keys, not just `shift` & `space`.
